src/QAT/qat/purr/integrations/qasm.py

```diff
@@ -1162,6 +1161,9 @@ def _walk_to_single_value(node: Tree):
 
                 if isinstance(registers, QubitRegister):
                     reg_index = self.transform_to_value(node.children[1])
+                    qubit = None
+                    if len(qubits:= registers.qubits) > reg_index:
```

Comment:
If the index of the qubit is greater than the size of the current list then directly index (and probably always exception)?

A nieve read says this may not be triggered, but if it was what's this give over the existing `next(...)`?